tests/test_bgp_af.rb

```diff
@@ -519,103 +756,146 @@ def test_dampening
-    assert_equal(2, bgp_af.dampening_reuse_time,
-                 'The wrong dampening reuse_time value is configured')
-    assert_equal(3, bgp_af.dampening_suppress_time,
-                 'The wrong dampening suppress_time value is configured')
-    assert_equal(4, bgp_af.dampening_max_suppress_time,
-                 'The wrong dampening max_suppress_time value is configured')
-    assert_empty(bgp_af.dampening_routemap,
-                 'A routemap should not be configured')
+    if platform == :nexus
+      # Check properties got set
+      af_string = get_bgp_af_dampening_params(asn, vrf, af)
+
+      pattern_params1 = 'Half-life time                 : 1 mins'
+      pattern_params2 = 'Suppress penalty               : 3'
+      pattern_params3 = 'Reuse penalty                  : 2'
+      pattern_params4 = 'Max suppress time              : 4 mins'
+      pattern_params5 = 'Max suppress penalty           : 32'
+
+      error = ("Error: 'dampening' properties are incorrect")
+
+      assert_match(pattern_params1, af_string, error)
+      assert_match(pattern_params2, af_string, error)
+      assert_match(pattern_params3, af_string, error)
+      assert_match(pattern_params4, af_string, error)
+      assert_match(pattern_params5, af_string, error)
+
+      # Check getters
+      assert_equal(bgp_af.dampening, %w(1 2 3 4),
+                   'Error: dampening getter did not match')
+      assert_equal(1, bgp_af.dampening_half_time,
+                   'The wrong dampening half_time value is configured')
+      assert_equal(2, bgp_af.dampening_reuse_time,
+                   'The wrong dampening reuse_time value is configured')
+      assert_equal(3, bgp_af.dampening_suppress_time,
+                   'The wrong dampening suppress_time value is configured')
+      assert_equal(4, bgp_af.dampening_max_suppress_time,
+                   'The wrong dampening max_suppress_time value is configured')
+      assert_empty(bgp_af.dampening_routemap,
+                   'A routemap should not be configured')
+    end
 
     #
     # Unset and verify
     #
     bgp_af.dampening = nil
 
-    af_string = get_bgp_af_cfg(asn, vrf, af)
-    pattern = /^ *dampening$/
+    if platform == :nexus
+      af_string = get_bgp_af_cfg(asn, vrf, af)
+      pattern = /^ *dampening$/
+    elsif platform == :ios_xr
+      af_string = @device.cmd("show run router bgp #{asn} " \
+                              'address-family ipv4 unicast bgp dampening')
+      pattern = /^ *bgp dampening$/
+    end
 
     refute_match(pattern, af_string, "Error: 'dampening' is still configured")
 
     #############################################
     # Set and verify 'dampening' with route-map #
     #############################################
-    vrf = 'brown'
-    af = %w(ipv6 unicast)
+    if platform == :nexus
+      vrf = 'brown'
+      af = %w(ipv6 unicast)
+    elsif platform == :ios_xr
+      vrf = 'default'
+      af = %w(ipv4 unicast)
```

Comment:
Agree. I changed it back to ipv6 unicast on both nexus and xr.  
Don’t remember why I used ipv4 on xr.  It also works with ipv6.

> On 30 Nov 2015, at 22:50, Glenn Matthews <notifications@github.com> wrote:
> 
> In tests/test_bgp_af.rb <https://github.com/cisco/cisco-network-node-utils/pull/87#discussion_r46208975>:
> 
> >  
> >      refute_match(pattern, af_string, "Error: 'dampening' is still configured")
> >  
> >      #############################################
> >      # Set and verify 'dampening' with route-map #
> >      #############################################
> > -    vrf = 'brown'
> > -    af = %w(ipv6 unicast)
> > +    if platform == :nexus
> > +      vrf = 'brown'
> > +      af = %w(ipv6 unicast)
> > +    elsif platform == :ios_xr
> > +      vrf = 'default'
> > +      af = %w(ipv4 unicast)
> does XR not support ipv6 unicast af? Or why the discrepancy?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/cisco/cisco-network-node-utils/pull/87/files#r46208975>.
>